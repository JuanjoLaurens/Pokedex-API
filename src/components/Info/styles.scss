@import '../../scss/bootstrap';

.pd-info {
  &__image {
    height: 300px;
    width: auto;

    @media only screen and (max-width: map-get($map: $grid-breakpoints, $key: lg)) {
      height: 200px;
    }
  }

  &__text {
    font-size: $h4-font-size;
  }
}
