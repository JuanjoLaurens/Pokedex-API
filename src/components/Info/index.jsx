import PropTypes from 'prop-types';
import cn from 'classnames/bind';
import React, { memo } from 'react';

import styles from './styles.scss';

const cx = cn.bind(styles);

const Info = ({ image, text }) => {
  return (
    <div
      className={cx('pd-info w-100 d-flex flex-column justify-content-center align-items-center')}
    >
      <img src={image} className={cx('pd-info__image rounded border my-5')} alt="info" />
      <p className={cx('pd-info__text text-muted mt-3')}>{text}</p>
    </div>
  );
};

Info.propTypes = {
  image: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default memo(Info);
