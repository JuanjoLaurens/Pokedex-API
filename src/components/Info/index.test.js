import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Info from './index';

const EXPECTED_TEXT = 'Lorem';

const makeProps = ({ image, text } = {}) => ({
  image: image || null,
  text: text || EXPECTED_TEXT,
});

const renderComponent = (props) => {
  return shallow(<Info {...props} />);
};

describe('Info', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = renderComponent(makeProps());
  });

  it('renders correctly according to snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders passed text', () => {
    expect(wrapper.find('p').text()).toEqual(EXPECTED_TEXT);
  });
});
