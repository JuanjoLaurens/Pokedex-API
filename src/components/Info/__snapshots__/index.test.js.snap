// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders correctly according to snapshot 1`] = `
<div
  className="pd-info w-100 d-flex flex-column justify-content-center align-items-center"
>
  <img
    alt="info"
    className="pd-info__image rounded border my-5"
    src={null}
  />
  <p
    className="pd-info__text text-muted mt-3"
  >
    Lorem
  </p>
</div>
`;
